packages/components/src/toolbar-item/index.js

```diff
@@ -14,24 +14,31 @@ import warning from '@wordpress/warning';
  */
 import ToolbarContext from '../toolbar-context';
 
-function ToolbarItem( { children, ...props }, ref ) {
+function ToolbarItem( { children, as: Component, ...props }, ref ) {
```

Comment:
I think at some point we need to introduce consistency to our components. Some of them use "tagName", others "as". We kind maybe move to "as" as a bit nicer while maintaining fallbacks for BC.